train.py

```diff
@@ -0,0 +1,249 @@
+import time
+import uuid
+from typing import Callable, Iterable
+
+import hydra
+import spacy
+import torch
+import einops
+import torch.nn as nn
+from tqdm import tqdm
+import torchtext.vocab.vocab
+from torch.optim.lr_scheduler import LambdaLR
+from omegaconf.omegaconf import OmegaConf, DictConfig
+
+import wandb
+from data.batch import Batch
+from model import make_model
+from utils import TrainState, rate
+from modules.label_smoothing import LabelSmoothing
+from data.dataloaders import load_vocab, load_tokenizers, create_dataloaders
+
+
+@hydra.main(config_path="config", config_name="conf", version_base=None)
+def main(cfg: DictConfig):
+    # load tokenizers and vocab
+    spacy_de, spacy_en = load_tokenizers()
+    vocab_src, vocab_tgt = load_vocab(spacy_de, spacy_en, cfg.data_slice)
+    # Train a model
+    model, run_id = train_model(vocab_src, vocab_tgt, spacy_de, spacy_en, cfg, device=cfg.device)
+
+    # save weights to a file
+    file_path = f"models/{cfg.file_prefix}-{run_id}-final.pt"
+    torch.save(
+        {
+            "model_state_dict": model.state_dict(),
+            "cfg": OmegaConf.to_container(cfg, resolve=True),
+        },
+        file_path,
+    )
+
+    # print a run_id of the model
+    print(run_id)
+
+
+def initialize_wandb(cfg: DictConfig) -> str:
+    # initialize experiment on WandB with unique run id
+    run_id = str(uuid.uuid1())[:8]
+    wandb.init(
+        project=cfg.project,
+        name=f"{cfg.run}-{run_id}",
+        config=OmegaConf.to_container(cfg, resolve=True),
+    )
+    return run_id
+
+
+def train_model(
+    vocab_src: torchtext.vocab.Vocab,
+    vocab_tgt: torchtext.vocab.Vocab,
+    spacy_de: spacy.Language,
+    spacy_en: spacy.Language,
+    cfg: DictConfig,
+    device="cpu",
+) -> tuple[nn.Module, str]:
+    print(f"Train using {device}")
+    if device != "cpu":
+        torch.cuda.set_device(device)
+    run_id = initialize_wandb(cfg)
+
+    # Get the index for padding token
+    pad_idx = vocab_tgt["<blank>"]
+
+    # define model parameters and create the model
+    model = make_model(
+        len(vocab_src),
+        len(vocab_tgt),
+        n=cfg.model.n,
+        d_model=cfg.model.d_model,
+        d_ff=cfg.model.d_ff,
+        h=cfg.model.h,
+        dropout=cfg.model.dropout,
+    )
+    if device != "cpu":
+        model.cuda(device)
```

Comment:
```suggestion
    model.to(device)
```

There shouldn't be a need for if statements like this, using `.to(device)` should just work for every `nn.Module`.